core/src/main/java/org/apache/brooklyn/location/byon/ByonLocationResolver.java

```diff
@@ -108,6 +109,7 @@ protected ConfigBag extractConfig(Map<?,?> locationFlags, String spec, LocationR
         MutableMap<String, Object> defaultProps = MutableMap.of();
         defaultProps.addIfNotNull("user", user);
         defaultProps.addIfNotNull("port", port);
+        defaultProps.addIfNotNull(LocationConfigKeys.EXTENSIONS.getName(), config.get(LocationConfigKeys.EXTENSIONS));
```

Comment:
After looking in more details: This adds the `EXTENSIONS` key to the machine spec. If this is skipped, i.e. the machine locations don't have the key, then wouldn't they get it from the parent location through inheritance? Why is BYON any more special than jclouds in handling this case? Is there a test which covers this case (and fails if the line is dropped)?